### PR TITLE
Wip/hughsie/plugin sync

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: app/views.py [hunk 1]
```diff
@@ -259,27 +259,30 @@ def upload():
...
     try:
-        arc = _build_cab(fileitem.filename, data)
-    except cabarchive.CorruptionError as e:
+        if fileitem.filename.endswith('.cab'):
+            istream = Gio.MemoryInputStream.new_from_bytes(GLib.Bytes.new(data))
+            arc = GCab.Cabinet.new()
+            arc.load(istream)
+            arc.extract(None)
+        else:
+            arc = _build_cab(fileitem.filename, data)
```
Comment (superm1) [inline app/views.py]: At least when I was reading through this, it was confusing - but this is for handling alternate extensions.
Can you rename` _build_cab` to something like `_repackage_archive` and add some comments?

File: schema.sql [hunk 2]
```diff
@@ -141,3 +140,15 @@ INSERT INTO settings (config_key, config_value)
   VALUES ('server_warning', 'This is a test instance and may be broken at any time.');
 INSERT INTO settings (config_key, config_value)
   VALUES ('firmware_baseuri', 'https://fwupd.org/downloads/');
+INSERT INTO settings (config_key, config_value)
+  VALUES ('cdn_sync_uri', 'https://s3.amazonaws.com/lvfstestbucket');
```
Comment (superm1) [inline schema.sql]: I think you should add something to README.md about updating this when setting up the instance.

Comment (superm1) [reply schema.sql]: And actually this URI doesn't seem to be used in code, is that intentional?

File: app/views.py (hunk 1 above)
Comment (hughsie) [reply app/views.py]: Agreed, thanks.

File: schema.sql (hunk 2 above)
Comment (hughsie) [reply schema.sql]: It is unused, I've removed it. I've not added anything about changing the database values in the README as all the plugin settings can be trivially changed in the admin-only "Server Setting" UI.

**Comment by hughsie** [comment]
I think Python3 is a good idea. I'll have to check the gnupg status of python3, but with cabarchive gone it's actually now possible.

**Comment by hughsie** [comment]
I did a quick run of 2to3 and there's quite a few places where we need to be more careful with the encoding. I'll merge this, and work on the Py3 stuff more tomorrow. Thanks for the review.

**Pull request merged**